src/handler/src/commands/tags/tag.ts

```diff
@@ -127,22 +127,37 @@ export default class implements Command {
 
 	private async delete(message: Message, args: Args, locale: string) {
 		const name = args.many();
-		if (!name.length) {
-			throw new Error(i18next.t('command.tag.common.execute.name_missing', { lng: locale }));
+		const user = args.option('user');
+
+		if (name.length) {
+			const [deletedTag] = await this.sql<{ name: string }>`
+				delete from tags
+				where name = ${joinTokens(name)}
+					and guild_id = ${message.guild_id!}
+				returning name`;
+			// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+			if (!deletedTag) {
+				throw new Error(i18next.t('command.tag.common.execute.not_found', { lng: locale }));
+			}
+			return this.rest.post(`/channels/${message.channel_id}/messages`, {
+				content: i18next.t('command.tag.delete.execute.deleted', { name: deletedTag.name, lng: locale }),
+			});
+		} else if (user) {
+			const [deletedTag] = await this.sql<{ name: string }>`
```

Comment:
This would return potentially many tags, so destructuring here and checking if it's undefined isn't necessarily the idiomatic way to check if things were deleted.